containers/src/Data/Set/Internal.hs

```diff
@@ -1827,7 +1827,7 @@ powerSet :: Set a -> Set (Set a)
 powerSet xs0 = insertMin empty (foldr' step Tip xs0) where
   step x pxs = insertMin (singleton x) (insertMin x `mapMonotonic` pxs) `glue` pxs
 
--- | \(O(mn)\) (conjectured). Calculate the Cartesian product of two sets.
+-- | \(O(nm)\). Calculate the Cartesian product of two sets.
```

Comment:
Forgot to mention, the comment below about uncertainty of the implementation being optimal needs to adjusted. Let me know if we want to remove it or keep some part of it.

Reply:
I think the part about conjecture should be replaced by your proof. The simple alternative proving to be slow seems like valuable information for someone in the future who wonders why we do the complicated thing, so it should stay.

Reply:
Done. I didn't want to be too handwavy, so I added a little more detail. I hope it's not too verbose now. Sadly, math too doesn't look very nice in a comment.